contracts/utils/Governable.sol

```diff
@@ -63,30 +64,40 @@ contract Governable {
      * @dev Transfers ownership of the contract to a new account (`newOwner`).
      * Can only be called by the current owner.
      */
-    function transferOwnership(address newOwner) public onlyGovernor {
+    function transferOwnership(address newOwner, uint32 nonce) public onlyGovernor {
+        require(refreshNonce < nonce, "Invalid nonce");
+        require(nonce <= refreshNonce + 1, "Nonce must increment by 1");
         _transferOwnership(newOwner);
+        refreshNonce = nonce;
     }
 
     /**
      * @dev Transfers ownership of the contract to a new account (`newOwner`).
      * Can only be called by the current owner.
      */
-    function transferOwnershipWithSignature(address newOwner, bytes memory sig) public {
+    function transferOwnershipWithSignature(address newOwner, uint32 nonce, bytes memory sig) public {
+        require(refreshNonce < nonce, "Invalid nonce");
+        require(nonce <= refreshNonce + 1, "Nonce must increment by 1");
```

Comment:
We need to add checks that nonce doesn't increase tremendously (causing overflow attack).

Reply:
Oops I had it backwards. This looks good.